addons/portal/models/mail_thread.py

```diff
@@ -76,3 +76,11 @@ def _sign_token(self, pid):
         secret = self.env["ir.config_parameter"].sudo().get_param("database.secret")
         token = (self.env.cr.dbname, self[self._mail_post_token_field], pid)
         return hmac.new(secret.encode('utf-8'), repr(token).encode('utf-8'), hashlib.sha256).hexdigest()
+
+    def _get_parent_sign_token(self, pid):
+        """ Overridden in models which have M2o 'parent' field and can be shared on
+        either an individual basis or indirectly in a group via the M2o record.
+
+        :return: False or logical parent's _sign_token() result
+        """
+        return False
```

Comment:
Coucou @tde-banana-odoo 🙂 

FYI, we add that method, to be able to compare the project hash with the hash given in parameter if the portal got a url of a project shared and would like to send a message in the chatter of a task linked to that project shared. Since in project, when we share a project, we actually share the tasks inside that project. 🙂 

I hope it is good for you. 🤞🏼  

Bisoutas! 😘

Reply:
Maybe ``_portal_get_parent_hask_token`` ? Otherwise agreed, don't have other ideas anyway :)